C/decoder/decode.c

```diff
@@ -0,0 +1,200 @@
+#include "decode.h"
+
+#define NUMERIC_ASCII_START 48
+#define UPPER_ALPHA_ASCII_START 65
+#define LOWER_ALPHA_ASCII_START 97
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+static char chars[83] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz#$%*+,-.:;=?@[]^_{|}~";
+
+/*
+	Image processing functions start
+*/
+
+static float sRGBToLinear(int value) {
+	float v = (float)value / 255;
+	if(v <= 0.04045) return v / 12.92;
+	else return powf((v + 0.055) / 1.055, 2.4);
+}
+
+static float signPow(float value, float exp) {
+	return copysignf(powf(fabsf(value), exp), value);
+}
+
+static int linearTosRGB(float value) {
+	float v = fmaxf(0, fminf(1, value));
+	if(v <= 0.0031308) return v * 12.92 * 255 + 0.5;
+	else return (1.055 * powf(v, 1 / 2.4) - 0.055) * 255 + 0.5;
+}
+
+/*
+	Image processing functions end
+*/
+
+/*
+	decoder helper functions start
+*/
+
+static inline unsigned char clamp_to_ubyte(int * src) {
```

Comment:
- Prefer `uint8_t` (from `<stdint.h>`) over `unsigned char`. The former is guaranteed to be a byte and the latter is not.
- My understanding is that "clamp" means to return 255 if greater than 255 and 0 if less than 0. Therefore I find this function name misleading.

Reply:
1. Yeah you are right, maybe shifting to `unit8_t` makes code more understandable as well, because a pixel can be understood as an unsigned 8-bit integer.

2. Yes, there was an issue in the `clamp`, fixed it in the latest commit.